pkg/strategy/all_in_one.go

```diff
@@ -82,6 +84,9 @@ func newAllInOneStrategy(ctx context.Context, jaeger *v1.Jaeger) S {
 	if viper.GetString("platform") == v1.FlagPlatformOpenShift {
 		if q := route.NewQueryRoute(jaeger).Get(); nil != q {
 			c.routes = append(c.routes, *q)
+			if link := consolelink.Get(jaeger, q); link != nil {
```

Comment:
This whole block looks strange, as neither `route.NewQueryRoute` nor `consolelink.Get` return nils. I think it can be simplified, can't it? They might even return the values directly instead of pointers.

Reply:
Well, route.NewQueryRoute(jaeger).Get() could potentially return nil if ingress is not enable.https://github.com/jaegertracing/jaeger-operator/blob/fd64f15cdd13bbe8360ca9fad2178a06838e3101/pkg/route/query.go#L25

In the case of consoleLink Would we want to have a flag for disable/enable it? If not of course I can simplify those lines. :)

Reply:
If we don't have a query, do we want a console link? I have a feeling that this will not run cleanly in plain Kubernetes. It might work right now because this is shielded at the controller, though.

IMO, if the route is protecting itself by checking whether the platform is openshift, then the console link should do the same.

Reply:
This is what is happening here, isn't it? If there is no query, (if query returns nil) it won't create a link.  Indeed this is checking the platform two lines above. It check the platform, then try to create a `queryRoute`, if query router is successfully created, then try to create a `consoleLink`.